File: JAVA/LinearSearch.java
import java.util.Random;
import java.util.Scanner;

public class LinearSearch{

    // Search function (Linear search)
    
    public static boolean isPresent(int arr[], int n){

        for(int i=0; i<arr.length; i++){
            if(arr[i] ==n) return true;
        }

        return false;

    }

    // Function to generate array of random integers
    
    public static int[] genArray(int n){

        Random rand = new Random();
        int arr[] = new int[n];
        for(int i=0; i<n;i++){
            arr[i] = rand.nextInt(n);
        }

        return arr;

    }


    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int n=sc.nextInt();// number of elements
        int[] array= genArray(n);

        int search= sc.nextInt();//Element to search
        
        for(int i=0; i<n;i++) System.out.print(array[i]+" ");//print array
        System.out.println();

        if(isPresent(array,search)) System.out.println("Element Present");
        
        else System.out.println("Element absent");

    }
}

File: C/linearSearch.c
// Best Case:- O(1), worst case:- O(n), average case:- O((n+1)/2)
// Use Cases:- Linear search is generally preferred for smaller and random ordered datasets.
//             and is less efficient in the case of larger datasets

#include<stdio.h>

int main(){
    int len,i,found=0,data,pos;
    printf("Enter the number of items in the array:");
    scanf("%d", &len);
    int arr[len];
    for(i=0;i<len;i++){
        printf("Enter the element %d:",(i+1));
        scanf("%d", &arr[i]);
    }
    printf("Enter the number you want to search:");
    scanf("%d",&data);
    for(i=0;i<len;i++){
        if(arr[i]==data){
            found=1;
            pos=i;
            break;
        }
    }
    (!found) ? printf("Element Not Found") : printf("Element Present at array index %d",pos);
    return 0;
}